From 2277675d15bf462bb78e7328769bb2835320b6fa Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 18 Jun 2014 12:09:41 -0400
Subject: [drm] i915: make i915 less chatty on resume

Message-id: <1403093381-31787-1-git-send-email-rclark@redhat.com>
Patchwork-id: 84256
O-Subject: [RHEL7 drm] drm/i915: make i915 less chatty on resume
Bugzilla: 1086777
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>

RHBZ: #1086777
Upstream: n/a

We hit these WARN_ON()'s on resume from suspend, although they are
benign.

The issue is known upstream, and a proper fix is being worked on.  But
that will be more of a refactoring, and not something we can pull in
before rhel71.  So to cut down on spurrious bug reports, remove the
WARN_ON()'s.

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_ddi.c b/drivers/gpu/drm/i915/intel_ddi.c
index a069b5e..8a250fd 100644
--- a/drivers/gpu/drm/i915/intel_ddi.c
+++ b/drivers/gpu/drm/i915/intel_ddi.c
@@ -353,7 +353,6 @@ void intel_ddi_put_crtc_pll(struct drm_crtc *crtc)
   if (plls->spll_refcount == 0) {
    DRM_DEBUG_KMS("Disabling SPLL\n");
    val = I915_READ(SPLL_CTL);
-   WARN_ON(!(val & SPLL_PLL_ENABLE));
    I915_WRITE(SPLL_CTL, val & ~SPLL_PLL_ENABLE);
    POSTING_READ(SPLL_CTL);
   }
@@ -363,7 +362,6 @@ void intel_ddi_put_crtc_pll(struct drm_crtc *crtc)
   if (plls->wrpll1_refcount == 0) {
    DRM_DEBUG_KMS("Disabling WRPLL 1\n");
    val = I915_READ(WRPLL_CTL1);
-   WARN_ON(!(val & WRPLL_PLL_ENABLE));
    I915_WRITE(WRPLL_CTL1, val & ~WRPLL_PLL_ENABLE);
    POSTING_READ(WRPLL_CTL1);
   }
@@ -373,7 +371,6 @@ void intel_ddi_put_crtc_pll(struct drm_crtc *crtc)
   if (plls->wrpll2_refcount == 0) {
    DRM_DEBUG_KMS("Disabling WRPLL 2\n");
    val = I915_READ(WRPLL_CTL2);
-   WARN_ON(!(val & WRPLL_PLL_ENABLE));
    I915_WRITE(WRPLL_CTL2, val & ~WRPLL_PLL_ENABLE);
    POSTING_READ(WRPLL_CTL2);
   }
-- 
1.7.1